agent/internal/agent.go

```diff
@@ -286,11 +286,13 @@ func (a *agent) tlsConfig() (*tls.Config, error) {
 	}
 
 	var certs []tls.Certificate
-	switch cert, err := a.Options.Security.TLS.ReadClientCertificate(); {
-	case err != nil:
-		return nil, errors.Wrap(err, "failed to read agent certificate file")
-	case cert != nil:
-		certs = append(certs, *cert)
+	if agentCert := a.Options.Security.TLS.ClientCert; agentCert != "" {
```

Comment:
I am somewhat groggy at the moment and a bit confused how it works. 
`ReadClientCertificate` itself already checks if `ClientCert == ""`: https://github.com/determined-ai/determined/blob/master/agent/internal/options.go#L112
and then it's supposed to `return nil, nil`, meaning this switch wouldn't hit any cases.
how does the same (?) check on top changes the code flow here?

Reply:
yeah I'm still looking into this, but the problem was the error "failed to read agent certificate file" and after the change it's not there...

Reply:
So I'm just more confused by the error and logically it makes no sense, so I think the best path right now is add logging and a precautionary string trimming step.